backend/layers/api/portal_api.py

```diff
@@ -234,15 +234,23 @@ def _collection_to_response(collection: CollectionVersionWithDatasets, access_ty
 
 def get_collection_details(collection_id: str, token_info: dict):
     """
-    Retrieves the collection information. Will look up for a published collection first,
-    and then looks up for a collection version
+    Retrieves the collection information. Will operate the following lookups, moving to the next one
+    only if the previous returns None.
+    Returns None only if no lookup was successful.
+    1. A published collection that has `collection_id` as the canonical id
+    2. A collection version with `collection_id` as the version_id (published or not)
+    3. An unpublished collection that has `collection_id` as the canonical id. This matches the case
+       where a collection doesn't have any published version yet, but we want to access it
+       via its permalink
     """
     # TODO: this logic might belong to the business layer?
     version = get_business_logic().get_published_collection_version(CollectionId(collection_id))
```

Comment:
`get_collection_version_from_canonical` already calls `get_published_collection_version` first, then tries to fetch a collection version from canonical if  `get_published_collection_version` is None. Order shouldn't matter here, so we can just switch this call to `get_collection_version_from_canonical` rather than adding a third call

Reply:
Good catch. I also changed the docstring to reflect the correct lookup order.